### PR TITLE
fix(platform): fix api differences

### PR DESCRIPTION
**Pull request by viceice**
Fix platform api differences. Now we have a full type check

### CODE REVIEW

**Comment by rarkins** [comment]
Is this *enforced* now? Or requires manual checking still?

**Comment by viceice** [comment]
it should be enforced now. but I would like to generate test code to fully enforce it automatically for all new platforms, like in #5500

File: lib/platform/azure/index.ts [hunk 1]
```diff
@@ -554,6 +554,8 @@ export async function ensureComment({
       'Comment is already update-to-date'
     );
   }
+
+  return true;
```
Comment (JamieMagee) [inline lib/platform/azure/index.ts]: `Promise.resolve(true);`?

Comment (viceice) [reply lib/platform/azure/index.ts]: not strictly nessessary, because we have a `async` function

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**